bindings/core/src/method_handler/utils.rs

```diff
@@ -123,6 +123,36 @@ pub(crate) fn call_utils_method_internal(method: UtilsMethod) -> Result<Response
 
             Response::TransactionFailureReason(context.validate()?)
         }
+        UtilsMethod::ManaWithDecay {
+            mana,
+            slot_index_created,
+            slot_index_target,
+            protocol_parameters,
+        } => Response::Amount(protocol_parameters.mana_with_decay(mana, slot_index_created, slot_index_target)?),
+        UtilsMethod::GenerateManaWithDecay {
+            amount,
+            slot_index_created,
+            slot_index_target,
+            protocol_parameters,
+        } => Response::Amount(protocol_parameters.generate_mana_with_decay(
+            amount,
+            slot_index_created,
+            slot_index_target,
+        )?),
+        UtilsMethod::OutputManaWithDecay {
+            output,
+            slot_index_created,
+            slot_index_target,
+            protocol_parameters,
+        } => {
+            let decayed_mana = output.decayed_stored_and_potential_mana(
```

Comment:
I'd put it directly inside the Response variant to be consistent with the others